pandas/tests/indexing/test_iloc.py

```diff
@@ -931,6 +931,14 @@ def test_iloc_setitem_td64_values_cast_na(self, value):
         expected = Series([NaT, 1, 2], dtype="timedelta64[ns]")
         tm.assert_series_equal(series, expected)
 
+    def test_setitem_mix_of_nan_and_interval(self):
```

Comment:
can you parameterize on all the cases in the OP

Reply:
Thanks, I parametrized the NA and the non-NA value, let me know if I should add more cases.